.eslintrc.json

```diff
@@ -0,0 +1,121 @@
+        "sourceType": "module",
+        "project": "./tsconfig.eslint.json"
+    },
+    "env": {
+        "browser": false,
+        "node": true,
+        "es6": true
+    },
+    "plugins": [
+        "@typescript-eslint", "jsdoc", "no-null", "import"
+    ],
+    "rules": {
+        "@typescript-eslint/adjacent-overload-signatures": "error",
+        "@typescript-eslint/array-type": "error",
+        "camelcase": "off",
+        "@typescript-eslint/camelcase": ["error", { "properties": "never", "allow": ["^[A-Za-z][a-zA-Za-z]+_[A-Za-z]+$"] }],
+        "@typescript-eslint/class-name-casing": "error",
+        "@typescript-eslint/consistent-type-definitions": ["error", "interface"],
+        "@typescript-eslint/interface-name-prefix": "error",
+        "@typescript-eslint/no-inferrable-types": "error",
+        "@typescript-eslint/no-misused-new": "error",
+        "@typescript-eslint/no-this-alias": "error",
+        "@typescript-eslint/no-unnecessary-qualifier": "error",
+        "@typescript-eslint/no-unnecessary-type-assertion": "error",
+        "@typescript-eslint/prefer-for-of": "error",
+        "@typescript-eslint/prefer-function-type": "error",
+        "@typescript-eslint/prefer-namespace-keyword": "error",
+        "semi": "off",
+        "@typescript-eslint/semi": "error",
+        "@typescript-eslint/triple-slash-reference": "error",
+        "@typescript-eslint/type-annotation-spacing": "error",
+        "@typescript-eslint/unified-signatures": "error",
+
+        // scripts/eslint/rules
+        "object-literal-surrounding-space": "error",
+        "no-type-assertion-whitespace": "error",
+        "type-operator-spacing": "error",
+        "only-arrow-functions": ["error", {
+            "allowNamedFunctions": true ,
+            "allowDeclarations": true
+        }],
+        "no-double-space": "error",
+        "boolean-trivia": "error",
+        "no-in-operator": "error",
+        "simple-indent": "error",
+        "debug-assert": "error",
+        "no-keywords": "error",
+
+        // eslint-plugin-import
+        "import/no-extraneous-dependencies": ["error", { "optionalDependencies": false }],
+
+        // eslint-plugin-no-null
+        "no-null/no-null": "error",
+
+        // eslint-plugin-jsdoc
+        "jsdoc/check-alignment": "error",
+
+        // eslint
+        "brace-style": ["error", "stroustrup", { "allowSingleLine": true }],
+        "constructor-super": "error",
+        "curly": ["error", "multi-line"],
+        "dot-notation": "error",
+        "eqeqeq": "error",
+        "linebreak-style": ["error", "windows"],
+        "new-parens": "error",
+        "no-caller": "error",
+        "no-duplicate-case": "error",
+        "no-duplicate-imports": "error",
+        "no-empty": "error",
+        "no-eval": "error",
+        "no-extra-bind": "error",
+        "no-fallthrough": "error",
+        "no-new-func": "error",
+        "no-new-wrappers": "error",
+        "no-redeclare": "error",
```

Comment:
Let's turn off `no-redeclare`. It doesn't work with namespace merging and is already covered by TypeScript itself.
We should also remove any associated `eslint-disable` comments for `no-redeclare`.